src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -556,6 +605,25 @@ protected static function get_file_path_from_theme( $file_name, $template = fals
 		return is_readable( $candidate ) ? $candidate : '';
 	}
 
+	protected static function has_same_blocks( $origin ) {
+		$registry = WP_Block_Type_Registry::get_instance();
+		$blocks   = $registry->get_all_registered();
+
+		// Is there metadata for all currently registered blocks?
+		$blocks = array_diff_key( $blocks, static::$blocks_meta[ $origin ] );
+		if ( empty( $blocks ) ) {
+			return false;
+		}
+
+		// Refill the cache.
+		static::$blocks_meta[ $origin ] = array();
+		foreach( $blocks as $block_name => $block_type ) {
```

Comment:
If all we care about is array keys, why not use the array_keys function.

Reply:
This is part a style choice, and part a minor performance nitpick. For performance: I want to do the less work in the lines before skipping (the path with most number of executions), and the current code requires one operation (`array_diff_key`):

```php
$block_diff = array_diff_key( $blocks, static::$blocks_meta[ $origin ] );
if ( empty( $block_diff ) ) {
    return true;
}
```

while the alternative (store the block names as values in `blocks_meta`) requires 3 operations (`array_diff` + `array_keys` + `sort`):


```php
$keys         = sort( array_keys( static::$blocks_meta[ $origin ] ) );
$block_diff = array_diff( $blocks, $keys );
if ( empty( $block_diff ) ) {
    return true;
}
```